libs/pika/executors/include/pika/executors/thread_pool_scheduler_queue_bypass.hpp

```diff
@@ -0,0 +1,275 @@
+
+            // invoke thread callable (call the actual function stored in the task)
+            PIKA_DETAIL_DP(bps_deb<5>,
+                debug(str<>("Execute"), threadinfo<thread_id_ref_type*>(&id), threaddata));
+            thread_result_type task_return = (*threaddata)(context_storage);
+
+            // did this task return a new task (ideally to be immediately switched into?)
+            thread_id_ref_type next = PIKA_MOVE(task_return.second);
+            if (next != nullptr)
+            {
+                PIKA_DETAIL_DP(bps_deb<5>,
+                    debug(str<>("Next task"), threadinfo<thread_id_ref_type*>(&next), threaddata));
+                // if next == id then we should just reschedule the task
+                // but lets assert to find out how often this happens
+                PIKA_ASSERT(next != id);
+
+                auto* scheduler = get_thread_id_data(next)->get_scheduler_base();
+                scheduler->schedule_thread(PIKA_MOVE(next),
+                    thread_schedule_hint(static_cast<std::int16_t>(thread_num)), true,
+                    thread_priority::boost);
+                scheduler->do_some_work(thread_num);
+            }
+
+            if (task_return.first == thread_schedule_state::terminated)
+            {
+                // just exit without any fanfare, terminated cleanup will take care of threaddata
+                PIKA_DETAIL_DP(bps_deb<5>,
+                    debug(str<>("Terminated"), threadinfo<thread_id_ref_type*>(&id), threaddata));
+                return;
+            }
+
+            // update our state with the new value returned from task execution
+            task_state =
+                thread_state(task_return.first, task_state.state_ex(), task_state.tag() + 1);
+
+            // if the threaddata->state still matches active_state update to new state
+            // (could be stolen/changed by another thread here?)
+            if (threaddata->restore_state(
+                    task_state, active_state, std::memory_order_relaxed, std::memory_order_relaxed))
+            {
+                switch (task_state.state())
+                {
+                case thread_schedule_state::pending_boost:
+                    threaddata->set_state(thread_schedule_state::pending);
+                    [[fallthrough]];
+                case thread_schedule_state::pending:
+                {
+                    data.scheduler_base->schedule_thread(id, thread_schedule_hint(thread_num),
+                        false /*allow_fallback*/, execution::thread_priority::high);
+                }
+                break;
+                case thread_schedule_state::suspended:
+                {
+                    std::cout << "thread_schedule_state::suspended" << std::endl;    //
+                    // thread sits in thread map until resumed
+                    return;
+                }
+                default: throw std::runtime_error("fix this thread state type");
```

Comment:
Can you update all of the plain throws to use `PIKA_THROW_EXCEPTION` with some appropriate error (`no_success` if nothing else).